packages/vite/src/utils/generator-util.spec.ts

```diff
@@ -0,0 +1,206 @@
+      );
+      expect(viteConfigPath).toEqual(
+        'apps/my-test-react-vite-app/vite.config.custom.ts'
+      );
+    });
+
+    it('should return correct path for vite.config file given a target name', () => {
+      const projectConfig = readProjectConfiguration(
+        tree,
+        'my-test-react-vite-app'
+      );
+      updateProjectConfiguration(tree, 'my-test-react-vite-app', {
+        ...projectConfig,
+        targets: {
+          ...projectConfig.targets,
+          'other-build': {
+            ...projectConfig.targets.build,
+            options: {
+              ...projectConfig.targets.build.options,
+              configFile: 'apps/my-test-react-vite-app/vite.other.custom.ts',
+            },
+          },
+        },
+      });
+
+      tree.write(`apps/my-test-react-vite-app/vite.other.custom.ts`, '');
+
+      const viteConfigPath = getViteConfigPathForProject(
+        tree,
+        'my-test-react-vite-app',
+        'other-build'
+      );
+      expect(viteConfigPath).toEqual(
+        'apps/my-test-react-vite-app/vite.other.custom.ts'
+      );
+    });
+  });
+
+  describe('findExistingTargetsInProject', () => {
+    it('should return the correct targets', () => {
+      mockReactAppGenerator(tree);
+      const { targets } = readProjectConfiguration(tree, 'my-test-react-app');
+
+      const existingTargets = findExistingTargetsInProject(targets);
+      expect(existingTargets).toMatchObject({
+        buildTarget: 'build',
+        serveTarget: 'serve',
+        testTarget: 'test',
+      });
+    });
+
+    it('should return the correct - undefined - targets for Angular apps', () => {
+      mockAngularAppGenerator(tree);
+      const { targets } = readProjectConfiguration(tree, 'my-test-angular-app');
+      const existingTargets = findExistingTargetsInProject(targets);
+      expect(existingTargets).toMatchObject({
+        buildTarget: undefined,
+        serveTarget: undefined,
+        testTarget: 'test',
+        unsupported: true,
+      });
+    });
+  });
+
+  describe('handleUnsupportedUserProvidedTargets', () => {
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+    it('should throw error if unsupported and user does not want to confirm', async () => {
+      const { Confirm } = require('enquirer');
+      const confirmSpy = jest.spyOn(Confirm.prototype, 'run');
+      confirmSpy.mockResolvedValueOnce(false);
+      expect.assertions(2);
+      const object = {
+        unsupportedUserProvidedTarget: {
+          build: true,
+        },
+        userProvidedTargets: {
+          build: 'my-build',
+          serve: 'my-serve',
+          test: 'my-test',
+        },
+        targets: {
+          build: 'build',
+          serve: 'serve',
+          test: 'test',
+        },
+      };
+
+      try {
+        await handleUnsupportedUserProvidedTargets(
+          object.unsupportedUserProvidedTarget,
+          object.userProvidedTargets,
+          object.targets
+        );
```

Comment:
This test would pass if no error is thrown.

Should add a `throw new Error('should not reach here')` or right after calling `handleUnsupportedUserProvidedTargets(...)`.